### PR TITLE
Complete typing of data module

### PR DESCRIPTION
**Pull request by jenshnielsen**

### CODE REVIEW

File: plottr/data/qcodes_dataset.py [hunk 1]
```diff
@@ -203,31 +203,34 @@ class QCodesDSLoader(Node):
...
+    def pathAndId(self, val: Tuple[Optional[str], Optional[int]]) -> None:
         if val != self.pathAndId:
             self._pathAndId = val
             self.nLoadedRecords = 0
 
     ### processing
-
-    def process(self, **kw):
+    # mypy does not correctly see the return annotation of pathAndId due to lack of support
+    # for decorated properties
```
Comment (astafan8) [inline plottr/data/qcodes_dataset.py]: should this comment be on line 218?

File: plottr/data/qcodes_dataset.py [hunk 2]
```diff
@@ -252,3 +255,5 @@ def process(self, **kw):
                 data.add_meta('qcodes_runTS', ds.run_timestamp())
                 self.nLoadedRecords = ds.number_of_results
                 return dict(dataOut=data)
+        else:
+            return None
```
Comment (astafan8) [inline plottr/data/qcodes_dataset.py]: hmm, don't we also need a `return None` for the `else` clause of the `if ds.number_of_results > self.nLoadedRecords:`?

**Review by astafan8 — APPROVED** [review]

File: plottr/data/qcodes_dataset.py (hunk 2 above)
Comment (jenshnielsen) [reply plottr/data/qcodes_dataset.py]: Yes you are right I missed that one

File: plottr/data/qcodes_dataset.py (hunk 1 above)
Comment (jenshnielsen) [reply plottr/data/qcodes_dataset.py]: Turns out this was related to the other one below

**Pull request merged**